examples/tree_1d_dgsem/elixir_advection_doublefloat.jl

```diff
@@ -17,12 +17,14 @@ solver = DGSEM(RealT = RealT, polydeg = 7, surface_flux = flux_lax_friedrichs)
 
 # CARE: Important to use higher precision datatype for coordinates
 # as these are used for type promotion of the mesh (points etc.)
-coordinates_min = (-RealT(1),) # minimum coordinate
-coordinates_max = (RealT(1),) # maximum coordinate
-cells_per_dimension = (256,)
+coordinates_min = -convert(RealT, 1) # minimum coordinate
+coordinates_max = convert(RealT, 1) # maximum coordinate
 
-# NOTE: StructuredMesh supports higher precision coordinates
-mesh = StructuredMesh(cells_per_dimension, coordinates_min, coordinates_max)
+# For `TreeMesh` the datatype has to be specified explicitly
+mesh = TreeMesh(coordinates_min, coordinates_max,
+                initial_refinement_level = 3,
+                n_cells_max = 30_000,
+                RealT = RealT)
```

Comment:
Can we please collect an issue listing these differences between the `TreeMesh` and the `StructuredMesh`?

Reply:
I added a comment in issue #1194 :

https://github.com/trixi-framework/Trixi.jl/issues/1194#issuecomment-2490503135